src/test/java/com/squareup/javawriter/builders/TypeSpecTest.java

```diff
@@ -33,22 +36,96 @@
             .addCode("return $S;\n", "taco")
             .build())
         .build();
-
-    // TODO: fix modifiers
-    // TODO: fix annotations
-
     assertThat(toString(taco)).isEqualTo(""
         + "package com.squareup.tacos;\n"
         + "\n"
+        + "import java.lang.Override;\n"
         + "import java.lang.String;\n"
         + "\n"
         + "class Taco {\n"
-        + "  String toString() {\n"
+        + "  @Override public final String toString() {\n"
         + "    return \"taco\";\n"
         + "  }\n"
         + "}\n");
   }
 
+  @Test public void anonymousInnerClass() throws Exception {
+    ClassName foo = ClassName.create("com.squareup.tacos", "Foo");
+    ClassName bar = ClassName.create("com.squareup.tacos", "Bar");
+    ClassName thingThang = ClassName.create(
+        "com.squareup.tacos", ImmutableList.of("Thing"), "Thang");
+    ParameterizedTypeName thingThangOfFooBar
+        = ParameterizedTypeName.create(thingThang, foo, bar);
+    ClassName thung = ClassName.create("com.squareup.tacos", "Thung");
+    ClassName simpleThung = ClassName.create("com.squareup.tacos", "SimpleThung");
+    ParameterizedTypeName thungOfSuperBar
+        = ParameterizedTypeName.create(thung, WildcardName.createWithLowerBound(bar));
+    ParameterizedTypeName thungOfSuperFoo
+        = ParameterizedTypeName.create(thung, WildcardName.createWithLowerBound(foo));
+    ParameterizedTypeName simpleThungOfBar = ParameterizedTypeName.create(simpleThung, bar);
+
+    ParameterSpec thungParameter = new ParameterSpec.Builder()
+        .addModifiers(Modifier.FINAL)
+        .type(thungOfSuperFoo)
+        .name("thung")
+        .build();
+    TypeSpec aSimpleThung = new TypeSpec.Builder()
+        .supertype(simpleThungOfBar)
+        .anonymousTypeArguments("$N", thungParameter)
```

Comment:
FYI: this is cool. Using $N to refer to a parameter automatically extracts its name, including name collision avoidance magic. So if the parameter was automatically renamed to "thung_1", then this would Do The Right Thing.